game_eggs/steamcmd_servers/subsistence/egg-subsistence-wine.json

```diff
@@ -0,0 +1,50 @@
+    "config": {
+        "files": "{}",
+        "startup": "{\r\n    \"done\": \"InternetCheckConnectionW\"\r\n}",
+        "logs": "{}",
+        "stop": "^C"
+    },
+    "scripts": {
+        "installation": {
+            "script": "apt -y update\r\napt -y --no-install-recommends install curl lib32gcc1 ca-certificates xz-utils\r\n\r\n## just in case someone removed the defaults.\r\nif [[ \"${STEAM_USER}\" == \"\" ]]; then\r\n    echo -e \"steam user is not set.\\n\"\r\n    echo -e \"Using anonymous user.\\n\"\r\n    STEAM_USER=anonymous\r\nelse\r\n    echo -e \"user set to ${STEAM_USER}\"\r\nfi\r\n\r\n## download and install steamcmd\r\ncd \/tmp\r\nmkdir -p \/mnt\/server\/steamcmd\r\ncurl -sSL -o steamcmd.tar.gz https:\/\/steamcdn-a.akamaihd.net\/client\/installer\/steamcmd_linux.tar.gz\r\ntar -xzvf steamcmd.tar.gz -C \/mnt\/server\/steamcmd\r\nmkdir -p \/mnt\/server\/steamapps # Fix steamcmd disk write error when this folder is missing\r\ncd \/mnt\/server\/steamcmd\r\n\r\n# SteamCMD fails otherwise for some reason, even running as root.\r\n# This is changed at the end of the install process anyways.\r\nchown -R root:root \/mnt\r\nexport HOME=\/mnt\/server\r\n\r\n## install game using steamcmd\r\n.\/steamcmd.sh +login ${STEAM_USER} +@sSteamCmdForcePlatformType windows +force_install_dir \/mnt\/server +app_update 1362640 ${INSTALL_FLAGS} validate +quit \r\n\r\n## set up 32 bit libraries\r\nmkdir -p \/mnt\/server\/.steam\/sdk32\r\ncp -v linux32\/steamclient.so ..\/.steam\/sdk32\/steamclient.so\r\n\r\n## set up 64 bit libraries\r\nmkdir -p \/mnt\/server\/.steam\/sdk64\r\ncp -v linux64\/steamclient.so ..\/.steam\/sdk64\/steamclient.so\r\n\r\n## install wine and winetricks\r\napt install -y --install-recommends wine wine64 xvfb cabextract\r\n\r\nmkdir -p \/mnt\/server\/.wine\r\n\r\n\r\ncurl -sSL https:\/\/raw.githubusercontent.com\/Winetricks\/winetricks\/master\/src\/winetricks -o \/usr\/sbin\/winetricks\r\n\r\ncurl https:\/\/dl.winehq.org\/wine\/wine-mono\/6.0.0\/wine-mono-6.0.0-x86.msi -o \/tmp\/wine-mono-6.0.0-x86.msi\r\n\r\nwine msiexec \/i \/tmp\/wine-mono-6.0.0-x86.msi\r\n\r\nchmod +x \/usr\/sbin\/winetricks\r\n\r\nexport WINEPREFIX=\/mnt\/server\/.wine\r\nexport WINEDLLOVERRIDES=\"mscoree,mshtml=\"\r\nexport DISPLAY=:0\r\nexport DISPLAY_WIDTH=1024\r\nexport DISPLAY_HEIGHT=768\r\nexport DISPLAY_DEPTH=16\r\nexport AUTO_UPDATE=1\r\nexport XVFB=1\r\n\r\nxvfb-run --auto-servernum winetricks -q --force win7 vcrun6 vcrun2019 corefonts d3dcompiler_43 d3dx9 dotnet40",
+            "container": "debian:buster-slim",
+            "entrypoint": "bash"
+        }
+    },
+    "variables": [
+        {
+            "name": "STEAM_USER",
+            "description": "",
+            "env_variable": "STEAM_USER",
+            "default_value": "anonymous",
+            "user_viewable": false,
+            "user_editable": true,
+            "rules": "required|string"
+        },
+        {
+            "name": "INSTALL_FLAGS",
```

Comment:
You should provide the SRCD_APPID and AUTO_UPDATE variables for the installation and auto-update. The INSTALL_FLAGS only applies to the install script and is currently ignored in the docker entrypoint - something we should take care of.

These are covered in our example [steamcmd install script](https://github.com/parkervcp/eggs/blob/master/scripts/steamcmd_installer.sh)